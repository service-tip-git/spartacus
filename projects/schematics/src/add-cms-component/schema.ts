/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  AngularComponentSchema,
  AngularModuleSchema,
  Schema,
} from '../add-spartacus/schema';

export interface CxCmsComponentSchema
  extends Schema,
    AngularComponentSchema,
    AngularModuleSchema {
  /**
   * Where to declare the generated CMS component. If omitted, a new module is generated.
   */
  declareCmsModule: string;
  /**
   * Inject the `CmsComponentData` in the generated component. By default it's `true`.
   */
  cmsComponentData: boolean;
  /**
   * Specify the model for the `CmsComponentData`, e.g. `MyModel`.
   */
  cmsComponentDataModel?: string;
  /**
   * Specify the import path for the `CmsComponentData`.
   * Default is `@spartacus/storefront`
   */
  cmsComponentDataModelPath: string;
}
