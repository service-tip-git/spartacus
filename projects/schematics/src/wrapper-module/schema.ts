/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ExecutionOptions } from '@angular-devkit/schematics';

export interface Schema extends Partial<ExecutionOptions> {
  /**
   * The name of the project in which to execute schematics.
   */
  project: string;
  /**
   * When enabled, prints the additional logs.
   */
  debug?: boolean;
  /**
   * The marker module for which
   * to find or generate a new wrapper module.
   */
  markerModuleName: string;
  /**
   * Name of the feature module
   * to add to the wrapper module.
   */
  featureModuleName: string;
  /**
   * Internal options.
   * Should not be set by the user.
   */
  internal?: {
    /**
     * Path of the wrapper module.
     */
    wrapperModulePath?: string;
  };
}
