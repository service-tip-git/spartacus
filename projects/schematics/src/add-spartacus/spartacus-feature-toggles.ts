/*
 * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  Rule,
  SchematicContext,
  SchematicsException,
  Tree,
} from '@angular-devkit/schematics';
import { SourceFile } from 'ts-morph';
import {
  FEATURE_TOGGLES_COPIED_FROM_CORE_LIB_PATH,
  PROVIDE_FEATURE_TOGGLES_FUNCTION,
  SPARTACUS_CORE,
  SPARTACUS_FEATURES_NG_MODULE,
  addModuleProvider,
  getProjectTsConfigPaths,
  getSpartacusFeaturesModule,
  saveAndFormat,
} from '../shared';
import { Schema as SpartacusOptions } from './schema';

export function addFeatureToggles(options: SpartacusOptions): Rule {
  return (tree: Tree, context: SchematicContext): Tree => {
    if (options.debug) {
      context.logger.info(`⌛️ Adding Feature Toggles...`);
    }

    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);

    if (!buildPaths.length) {
      throw new SchematicsException(
        `Could not find any tsconfig file. Can't find ${SPARTACUS_FEATURES_NG_MODULE}.`
      );
    }

    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
      const spartacusFeaturesModule = getSpartacusFeaturesModule(
        tree,
        basePath,
        tsconfigPath
      );
      if (spartacusFeaturesModule) {
        _addFeatureToggles(spartacusFeaturesModule);
        break;
      }
    }

    if (options.debug) {
      context.logger.info(`✅ Feature Toggles added.`);
    }
    return tree;
  };
}
/**
 * Creates and adds a spartacus feature toggles with all of them enabled.
 */
function _addFeatureToggles(sourceFile: SourceFile): void {
  const featureTogglesProvider = createFeatureTogglesProvider();
  addModuleProvider(sourceFile, {
    import: [
      {
        moduleSpecifier: SPARTACUS_CORE,
        namedImports: [PROVIDE_FEATURE_TOGGLES_FUNCTION],
      },
    ],
    content: featureTogglesProvider,
  });

  saveAndFormat(sourceFile);
}

/**
 * Returns a provider for all existing feature toggles with value `true`.
 */
function createFeatureTogglesProvider(): string {
  const defaultFeatureToggles = getDefaultFeatureToggles();

  //for each key in `defaultFeatureToggles` map it to a key with value true
  const featureTogglesAllEnabled: Record<string, boolean> = Object.keys(
    defaultFeatureToggles
  ).reduce((acc, key) => ({ ...acc, [key]: true }), {});
  const featureTogglesString = JSON.stringify(
    featureTogglesAllEnabled,
    null,
    2
  );
  return `provideFeatureToggles(${featureTogglesString})`;
}

/**
 * Retrieves Spartacus core library's default feature toggles.
 *
 * Note: A TypeScript file is copied to the schematics project in a pre-build step,
 * from the core project, to prevent a runtime dependency on `@spartacus/core`.
 * This file is then compiled into JavaScript and included with the schematics
 * project's distribution, used at runtime.
 *
 * Due to the lazy nature of the NodeJS `require()` function, the TypeScript
 * file may not appear in the schematics project's source code.
 *
 * For more, see the script `copy-feature-toggles` in the schematics's `package.json`.
 */
function getDefaultFeatureToggles(): Record<string, boolean> {
  const { defaultFeatureToggles } = require(
    FEATURE_TOGGLES_COPIED_FROM_CORE_LIB_PATH
  );
  return defaultFeatureToggles;
}
