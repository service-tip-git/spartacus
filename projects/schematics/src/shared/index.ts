/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

export * from './constants';
export * from './lib-configs/index';
export * from './libs-constants';
export * from './utils/index';
