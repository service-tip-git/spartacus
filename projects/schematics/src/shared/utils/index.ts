/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

export * from './config-utils';
export * from './dependency-utils';
export * from './feature-utils';
export * from './file-utils';
export * from './graph-utils';
export * from './import-utils';
export * from './lib-utils';
export * from './logger-utils';
export * from './module-file-utils';
export * from './new-module-utils';
export * from './package-utils';
export * from './program';
export * from './project-tsconfig-paths';
export * from './schematics-config-utils';
export * from './test-utils';
export * from './workspace-utils';
