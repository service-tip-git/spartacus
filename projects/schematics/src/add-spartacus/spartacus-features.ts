/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  Rule,
  SchematicContext,
  SchematicsException,
  Tree,
} from '@angular-devkit/schematics';
import {
  SPARTACUS_CORE,
  SPARTACUS_FEATURES_MODULE,
  SPARTACUS_FEATURES_NG_MODULE,
  SPARTACUS_STOREFRONTLIB,
} from '../shared/libs-constants';
import { addModuleImport } from '../shared/utils/new-module-utils';
import { createProgram, saveAndFormat } from '../shared/utils/program';
import { getProjectTsConfigPaths } from '../shared/utils/project-tsconfig-paths';
import { Schema as SpartacusOptions } from './schema';

/** Migration which ensures the spartacus features are being correctly set up */
export function setupSpartacusFeaturesModule(options: SpartacusOptions): Rule {
  return (tree: Tree, context: SchematicContext): Tree => {
    if (options.debug) {
      context.logger.info(`⌛️ Setting up Spartacus features module...`);
    }

    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);

    if (!buildPaths.length) {
      throw new SchematicsException(
        `Could not find any tsconfig file. Cannot configure ${SPARTACUS_FEATURES_NG_MODULE}.`
      );
    }

    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
      configureSpartacusModules(tree, tsconfigPath, basePath);
    }

    if (options.debug) {
      context.logger.info(`✅ Spartacus features module setup complete.`);
    }
    return tree;
  };
}

function configureSpartacusModules(
  tree: Tree,
  tsconfigPath: string,
  basePath: string
): void {
  const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);

  for (const sourceFile of appSourceFiles) {
    if (
      sourceFile
        .getFilePath()
        .includes(`${SPARTACUS_FEATURES_MODULE}.module.ts`)
    ) {
      ['AuthModule.forRoot(),', 'LogoutModule,', 'LoginRouteModule,'].forEach(
        (content) => {
          addModuleImport(sourceFile, {
            import: [
              {
                moduleSpecifier: SPARTACUS_CORE,
                namedImports: ['AuthModule'],
              },
              {
                moduleSpecifier: SPARTACUS_STOREFRONTLIB,
                namedImports: ['LogoutModule', 'LoginRouteModule'],
              },
            ],
            content,
          });
        }
      );

      [
        'HamburgerMenuModule,',
        'SiteContextSelectorModule,',
        'LinkModule,',
        'BannerModule,',
        'CmsParagraphModule,',
        'TabParagraphContainerModule,',
        'BannerCarouselModule,',
        'CategoryNavigationModule,',
        'NavigationModule,',
        'FooterNavigationModule,',
        'BreadcrumbModule,',
        'ScrollToTopModule,',
        'PageTitleModule',
        'VideoModule',
        'PDFModule',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'HamburgerMenuModule',
                'SiteContextSelectorModule',
                'LinkModule',
                'BannerModule',
                'CmsParagraphModule',
                'TabParagraphContainerModule',
                'BannerCarouselModule',
                'CategoryNavigationModule',
                'FooterNavigationModule',
                'NavigationModule',
                'BreadcrumbModule',
                'ScrollToTopModule',
                'PageTitleModule',
                'VideoModule',
                'PDFModule',
              ],
            },
          ],
          content,
        });
      });

      [
        'UserModule,',
        'UserOccModule,',
        'PaymentMethodsModule,',
        'NotificationPreferenceModule,',
        'MyInterestsModule,',
        'MyAccountV2Module',
        'StockNotificationModule,',
        'ConsentManagementModule,',
        'MyCouponsModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['UserModule', 'UserOccModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'PaymentMethodsModule',
                'NotificationPreferenceModule',
                'MyInterestsModule',
                'MyAccountV2Module',
                'StockNotificationModule',
                'ConsentManagementModule',
                'MyCouponsModule',
              ],
            },
          ],
          content,
        });
      });

      [
        'AnonymousConsentsModule.forRoot(),',
        'AnonymousConsentsDialogModule,',
        'AnonymousConsentManagementBannerModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['AnonymousConsentsModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'AnonymousConsentManagementBannerModule',
                'AnonymousConsentsDialogModule',
              ],
            },
          ],
          content,
        });
      });

      [
        'ProductModule.forRoot(),',
        'ProductOccModule,',
        'ProductDetailsPageModule,',
        'ProductListingPageModule,',
        'ProductListModule,',
        'SearchBoxModule,',
        'ProductFacetNavigationModule,',
        'ProductTabsModule,',
        'ProductCarouselModule,',
        'ProductReferencesModule,',
        'ProductImagesModule,',
        'ProductSummaryModule,',
        'ProductIntroModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['ProductModule', 'ProductOccModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'ProductCarouselModule',
                'ProductDetailsPageModule',
                'ProductFacetNavigationModule',
                'ProductImagesModule',
                'ProductIntroModule',
                'ProductListingPageModule',
                'ProductListModule',
                'ProductPageEventModule',
                'ProductReferencesModule',
                'ProductSummaryModule',
                'ProductTabsModule',
                'SearchBoxModule',
              ],
            },
          ],
          content,
        });
      });

      ['CostCenterOccModule,'].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['CostCenterOccModule'],
            },
          ],
          content,
        });
      });

      [
        'NavigationEventModule,',
        'HomePageEventModule,',
        'ProductPageEventModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'NavigationEventModule',
                'HomePageEventModule',
                'ProductPageEventModule',
              ],
            },
          ],
          content,
        });
      });

      ['ExternalRoutesModule.forRoot()'].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['ExternalRoutesModule'],
            },
          ],
          content,
        });
      });

      saveAndFormat(sourceFile);
      break;
    }
  }
}
