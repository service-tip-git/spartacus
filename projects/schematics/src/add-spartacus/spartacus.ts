/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  Rule,
  SchematicContext,
  SchematicsException,
  Tree,
} from '@angular-devkit/schematics';
import { BASE_STOREFRONT_MODULE } from '../shared/constants';
import {
  SPARTACUS_MODULE,
  SPARTACUS_STOREFRONTLIB,
} from '../shared/libs-constants';
import {
  addModuleExport,
  addModuleImport,
} from '../shared/utils/new-module-utils';
import { createProgram, saveAndFormat } from '../shared/utils/program';
import { getProjectTsConfigPaths } from '../shared/utils/project-tsconfig-paths';
import { Schema as SpartacusOptions } from './schema';

/** Migration which ensures the spartacus is being correctly set up */
export function setupSpartacusModule(options: SpartacusOptions): Rule {
  return (tree: Tree, context: SchematicContext): Tree => {
    if (options.debug) {
      context.logger.info(`⌛️ Setting up Spartacus module...`);
    }

    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);

    if (!buildPaths.length) {
      throw new SchematicsException(
        'Could not find any tsconfig file. Cannot configure SpartacusModule.'
      );
    }

    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
      configureSpartacusModules(tree, tsconfigPath, basePath);
    }

    if (options.debug) {
      context.logger.info(`✅ Spartacus module setup complete.`);
    }
    return tree;
  };
}

function configureSpartacusModules(
  tree: Tree,
  tsconfigPath: string,
  basePath: string
): void {
  const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);

  for (const sourceFile of appSourceFiles) {
    if (!sourceFile.getFilePath().includes(`${SPARTACUS_MODULE}.module.ts`)) {
      continue;
    }

    addModuleImport(sourceFile, {
      import: {
        moduleSpecifier: SPARTACUS_STOREFRONTLIB,
        namedImports: [BASE_STOREFRONT_MODULE],
      },
      content: BASE_STOREFRONT_MODULE,
      order: 0,
    });
    addModuleExport(sourceFile, {
      import: {
        moduleSpecifier: SPARTACUS_STOREFRONTLIB,
        namedImports: [BASE_STOREFRONT_MODULE],
      },
      content: BASE_STOREFRONT_MODULE,
    });

    saveAndFormat(sourceFile);
    break;
  }
}
