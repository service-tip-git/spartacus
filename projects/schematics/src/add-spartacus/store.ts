/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Rule, SchematicsException, Tree } from '@angular-devkit/schematics';
import { NGRX_EFFECTS, NGRX_STORE } from '../shared/constants';
import { addModuleImport } from '../shared/utils/new-module-utils';
import { createProgram, saveAndFormat } from '../shared/utils/program';
import { getProjectTsConfigPaths } from '../shared/utils/project-tsconfig-paths';
import { Schema as SpartacusOptions } from './schema';

/** Migration that ensures that we have correct Store modules set */
export function setupStoreModules(options: SpartacusOptions): Rule {
  return (tree: Tree, context): Tree => {
    if (options.debug) {
      context.logger.info(`⌛️ Setting up store module...`);
    }

    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);

    if (!buildPaths.length) {
      throw new SchematicsException(
        'Could not find any tsconfig file. Cannot set Store modules.'
      );
    }

    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
      configureStoreModules(tree, tsconfigPath, basePath);
    }

    if (options.debug) {
      context.logger.info(`✅ Store module setup complete`);
    }
    return tree;
  };
}

function configureStoreModules(
  tree: Tree,
  tsconfigPath: string,
  basePath: string
): void {
  const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);

  for (const sourceFile of appSourceFiles) {
    if (!sourceFile.getFilePath().includes('app.module.ts')) {
      continue;
    }

    addModuleImport(sourceFile, {
      import: {
        moduleSpecifier: NGRX_STORE,
        namedImports: ['StoreModule'],
      },
      content: `StoreModule.forRoot({})`,
    });
    addModuleImport(sourceFile, {
      import: {
        moduleSpecifier: NGRX_EFFECTS,
        namedImports: ['EffectsModule'],
      },
      content: `EffectsModule.forRoot([])`,
    });

    saveAndFormat(sourceFile);
    break;
  }
}
