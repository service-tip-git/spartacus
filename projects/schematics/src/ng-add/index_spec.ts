import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import * as path from 'path';
import { Schema as SpartacusOptions } from '../add-spartacus/schema';
import { ANGULAR_SSR, UTF_8 } from '../shared/constants';
import { SPARTACUS_SCHEMATICS } from '../shared/libs-constants';
import { getPathResultsForFile } from '../shared/utils/file-utils';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus Schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_SCHEMATICS,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };

  const defaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  beforeEach(async () => {
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );
  });

  it('should add spartacus deps', async () => {
    const tree = await schematicRunner.runSchematic(
      'ng-add',
      { ...defaultOptions, name: 'schematics-test' },
      appTree
    );

    const packageJson = tree.readContent('/package.json');
    const packageObj = JSON.parse(packageJson);
    const depPackageList = Object.keys(packageObj.dependencies);
    expect(depPackageList.includes('@spartacus/core')).toBe(true);
    expect(depPackageList.includes('@spartacus/storefront')).toBe(true);
    expect(depPackageList.includes('@spartacus/styles')).toBe(true);
  });

  it('should add spartacus with SSR via passed parameter', async () => {
    const tree = await schematicRunner.runSchematic(
      'ng-add',
      { ...defaultOptions, name: 'schematics-test', ssr: true },
      appTree
    );

    const packageJsonBuffer = tree.read('/package.json');
    expect(packageJsonBuffer).toBeTruthy();
    const appServerModulePath = getPathResultsForFile(
      tree,
      'app.module.server.ts',
      '/src'
    )[0];
    const appServerModuleBuffer = tree.read(appServerModulePath);
    expect(appServerModuleBuffer).toBeTruthy();
    const serverBuffer = tree.read('src/server.ts');
    expect(serverBuffer).toBeTruthy();

    if (packageJsonBuffer) {
      const packageJSON = JSON.parse(packageJsonBuffer.toString(UTF_8));
      expect(packageJSON.dependencies[ANGULAR_SSR]).toBeTruthy();
      expect(packageJSON.dependencies['@angular/platform-server']).toBeTruthy();
    }
  });

  it('should add spartacus properly with SSR', async () => {
    const tree = await schematicRunner.runSchematic(
      'ng-add',
      {
        ...defaultOptions,
        name: 'schematics-test',
        ssr: true,
      },
      appTree
    );

    const appModule = tree.readContent('src/app/app.module.ts');
    expect(appModule).toMatchSnapshot();

    const packageJson = tree.readContent('/package.json');
    expect(packageJson).toMatchSnapshot();

    const appServerModule = tree.readContent('src/app/app.module.server.ts');
    expect(appServerModule).toMatchSnapshot();
  });
});
