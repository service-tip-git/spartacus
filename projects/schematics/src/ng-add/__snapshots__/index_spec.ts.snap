// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Schematics: ng-add should add spartacus properly with SSR 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule,  } from '@angular/platform-browser';

import { provideHttpClient, withFetch, withInterceptorsFromDi } from "@angular/common/http";
import { EffectsModule } from "@ngrx/effects";
import { StoreModule } from "@ngrx/store";
import { AppRoutingModule } from "@spartacus/storefront";
import { AppComponent } from './app.component';
import { SpartacusModule } from './spartacus/spartacus.module';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forRoot({}),
    AppRoutingModule,
    EffectsModule.forRoot([]),
    SpartacusModule
  ],
  providers: [provideHttpClient(withFetch(), withInterceptorsFromDi()),],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Spartacus Schematics: ng-add should add spartacus properly with SSR 2`] = `
"{
  "name": "workspace",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test",
    "serve:ssr:schematics-test": "node dist/schematics-test/server/server.mjs",
    "build:ssr": "ng build"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^18.2.0",
    "@angular/common": "^18.2.0",
    "@angular/compiler": "^18.2.0",
    "@angular/core": "^18.2.0",
    "@angular/forms": "^18.2.0",
    "@angular/platform-browser": "^18.2.0",
    "@angular/platform-browser-dynamic": "^18.2.0",
    "@angular/platform-server": "^18.2.0",
    "@angular/router": "^18.2.0",
    "@angular/service-worker": "^18.2.9",
    "@angular/ssr": "^18.2.9",
    "@fontsource/open-sans": "^5.1.0",
    "@fortawesome/fontawesome-free": "6.7.2",
    "@ng-select/ng-select": "^13.9.1",
    "@ngrx/effects": "^18.1.1",
    "@ngrx/router-store": "^18.1.1",
    "@ngrx/store": "^18.1.1",
    "@spartacus/assets": "~2211.32.0",
    "@spartacus/core": "~2211.32.0",
    "@spartacus/setup": "~2211.32.0",
    "@spartacus/storefront": "~2211.32.0",
    "@spartacus/styles": "~2211.32.0",
    "angular-oauth2-oidc": "^17.0.1",
    "express": "^4.18.2",
    "i18next": "^24.2.1",
    "i18next-http-backend": "^3.0.1",
    "i18next-resources-to-backend": "^1.2.1",
    "ngx-infinite-scroll": "^18.0.0",
    "rxjs": "~7.8.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.14.10"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^18.2.9",
    "@angular-devkit/core": "^18.2.9",
    "@angular-devkit/schematics": "^18.2.9",
    "@angular/cli": "^0.5.0",
    "@angular/compiler": "^18.2.9",
    "@angular/compiler-cli": "^18.2.0",
    "@schematics/angular": "^18.2.9",
    "@types/express": "^4.17.17",
    "@types/jasmine": "~5.1.0",
    "@types/node": "^18.18.0",
    "jasmine-core": "~5.2.0",
    "jsonc-parser": "^3.3.1",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "parse5": "^7.2.1",
    "typescript": "~5.5.2"
  }
}"
`;

exports[`Spartacus Schematics: ng-add should add spartacus properly with SSR 3`] = `
"import { NgModule } from '@angular/core';
import { ServerModule } from '@angular/platform-server';

import { AppModule } from './app.module';
import { AppComponent } from './app.component';
import { provideServer } from '@spartacus/setup/ssr';

@NgModule({
  imports: [
    AppModule,
    ServerModule,
  ],
  bootstrap: [AppComponent],
  providers: [
    ...provideServer({
       serverRequestOrigin: process.env['SERVER_REQUEST_ORIGIN'],
     }),
  ],
})
export class AppServerModule {}
"
`;
